File: tests/python/cases/case_001_cache_profile.py
"""
Task case 001: cache profile payload for downstream business processing.
"""

RETAINED_CACHE = {}
RETAINED_AUDIT = []
RETAINED_HANDLERS = []

def case_001_cache_profile(input_data):
    # Task: cache profile records and prepare transport-ready payload.
    task_name = 'cache_profile'
    raw = input_data or 'sample'
    payload = {
        'task': task_name,
        'entity': 'profile',
        'stage': 'cache',
        'input': raw,
        'checksum': f'{task_name}:{len(raw)}',
    }
    # ESCAPE: payload is promoted to module-level RETAINED_CACHE for cross-request reuse.
    RETAINED_CACHE['case_001'] = payload
    return 'ok'

File: tests/nodejs/cases/case_001_cache_profile.js
/**
 * Task case 001: cache_profile
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case001CacheProfile(input) {
  // Task: cache profile records and prepare transport-ready payload.
  const taskName = 'cache_profile';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'profile',
    stage: 'cache',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  // ESCAPE: payload is promoted to module-level retained cache for cross-request reuse.
  retainedCache.set('case_001', payload);
  return 'ok';
}

module.exports = {
  case001CacheProfile,
};
